Add syntax for separate from MonadCombine.

peterneyens opened the pull request:
I based myself on the structure in syntax.bitraverse for the two implicit conversions (the
existing for unite and the new for separate).

MonadCombine separate was added in #864 by @adelbertc.

ceedubs commented:
:+1: looks good. Thanks!

adelbertc commented:
Hm what is the motivation of calling it "nested" and having it in a `MonadCombineSyntax1` as opposed to just the `MonadCombineSyntax`?

peterneyens commented:
I followed the same structure as in the [bitraverse syntax](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/syntax/bitraverse.scala).

`MonadCombineSyntax1` doesn't seem necessary yes (and neither does `BitraverseSyntax1` ?)

Should I rename `NestedMonadCombineOps` to `SeparateOps` (like `FlattenOps` and `IfMOps` in the [flatMap syntax](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/syntax/flatMap.scala)?

peterneyens commented:
I have renamed the `NestedMonadCombineOps` to `SeparateOps` and moved the implicit conversion into `MonadCombineSyntax`.

I have also added a test for `MonadCombine` in [SyntaxTests](https://github.com/typelevel/cats/blob/master/tests/src/test/scala/cats/tests/SyntaxTests.scala). Am I right that this file just needs to compile, for the tests to be valid ?

ceedubs commented:
👍 looks good to me. Thanks @peterneyens!

non commented:
Looks good to me. 👍
